ush/rocoto/setup_expt.py

```diff
@@ -1,76 +1,86 @@
 #!/usr/bin/env python3
 
-###############################################################
-# < next few lines under version control, D O  N O T  E D I T >
-# $Date$
-# $Revision$
-# $Author$
-# $Id$
-###############################################################
+'''
+Entry point for setting up an experiment in the global-workflow
+'''
 
 import os
-import sys
 import glob
 import shutil
-import socket
-from datetime import datetime, timedelta
+from datetime import datetime
 from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter
 import workflow_utils as wfu
 
-global expdir, configdir, comrot, pslot, resdet, resens, nens, cdump, idate, edate, gfs_cyc
 
+def makedirs_if_missing(dirname):
+    if not os.path.exists(dirname):
+        os.makedirs(dirname)
 
-def makedirs_if_missing(d):
-    if not os.path.exists(d):
-        os.makedirs(d)
 
+def fill_COMROT(host, inputs):
 
-def create_EXPDIR():
+    fill_modes = {
+        'cycled': fill_COMROT_cycled,
+        'forecast-only': fill_COMROT_forecasts
+    }
 
-    makedirs_if_missing(expdir)
-    configs = glob.glob(f'{configdir}/config.*')
-    if len(configs) == 0:
-        msg = f'no config files found in {configdir}'
-        raise IOError(msg)
-    for config in configs:
-        shutil.copy(config, expdir)
+    try:
+        fill_modes[inputs.mode](host, inputs)
+    except KeyError:
+        raise NotImplementedError(f'{mode} is not a supported mode.\n' +
+                                  'Currently supported modes are:\n' +
+                                  f'{" | ".join(fill_modes.keys())}')
 
     return
 
 
-def create_COMROT():
-
-    idatestr = idate.strftime('%Y%m%d%H')
-    cymd = idate.strftime('%Y%m%d')
-    chh = idate.strftime('%H')
+def fill_COMROT_cycled(host, inputs):
 
-    makedirs_if_missing(comrot)
+    idatestr = inputs.idate.strftime('%Y%m%d%H')
+    comrot = os.path.join(inputs.comrot, inputs.pslot)
 
     # Link ensemble member initial conditions
-    enkfdir = f'enkf{cdump}.{cymd}/{chh}'
+    enkfdir = f'enkf{inputs.cdump}.{idatestr[:8]}/{idatestr[8:]}'
     makedirs_if_missing(os.path.join(comrot, enkfdir))
-    for i in range(1, nens + 1):
-        makedirs_if_missing(os.path.join(comrot, enkfdir, f'mem{i:03d}'))
-        os.symlink(os.path.join(icsdir, idatestr, f'C{resens}', f'mem{i:03d}', 'INPUT'),
-                   os.path.join(comrot, enkfdir, f'mem{i:03d}', 'INPUT'))
+    for ii in range(1, inputs.nens + 1):
```

Comment:
Same argument applies to single character iterators.  I thought about `mem`, but then chose not to since `mem` is used as a string in this loop, and using `mem` as an iterator would be confusing.   I could have chosen `nn` instead.